airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -81,23 +81,33 @@ def backoff_time(self, response: requests.Response) -> Union[int, float]:
         return max(backoff_time, 60)  # This is a guarantee that no negative value will be returned.
 
     def read_records(self, stream_slice: Mapping[str, any] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+        # get out the stream_slice parts for later use.
+        organisation = stream_slice.get("organization", "")
+        repository = stream_slice.get("repository", "")
+        # Reading records while handling the errors
         try:
             yield from super().read_records(stream_slice=stream_slice, **kwargs)
         except HTTPError as e:
-            error_msg = str(e)
-
+            error_msg = str(e.response.json().get("message"))
```

Comment:
Are any probability that error will be not in json format?

Reply:
No, the response should be always in `json` format.